pkg/util/source/inspector.go

```diff
@@ -336,6 +336,11 @@ func (i *baseInspector) discoverKamelets(meta *Metadata) {
 }
 
 func (i *baseInspector) addDependencies(uri string, meta *Metadata, consumer bool) error {
+	if !strings.Contains(uri, ":") {
```

Comment:
Not sure if this statement is okey to be here. I mean, in theory we may expect an uri with the form `mycomponent`, without the scheme. I wonder if it makes sense: 1) to check for the specific use case (`{{}}`) directly in the `DecodeComponent()` func, 2) if it makes sense to have the check in the callers of `addDependencies()` func.

Reply:
the `DecodeComponent()` logic requires a scheme and a resource path part otherwise it will return nil (see https://github.com/apache/camel-k/blob/main/pkg/util/camel/camel_runtime_catalog.go#L185)

so the uri you have mentioned with just `mycomponent` will fail with the new error `component not found for uri ...`

Reply:
this has always been the logic but now we use to raise errors when there is a uri not matching the format `scheme:resourcepath`